drivers/periph_gpio: let gpio_read() return bool

maribu opened the pull request:
### Contribution description

Since https://github.com/RIOT-OS/RIOT/pull/20935 gpio_write() uses a `bool` instead of an `int`. This does the same treatment for `gpio_read()`.

This does indeed add an instruction to `gpio_read()` implementations. However, users caring about an instruction more are better served with `gpio_ll_read()` anyway. And `gpio_read() == 1` is often seen in newcomer's code, which would now work as expected.

### Testing procedure

This should not cause any regressions.

I cannot find any way to use `int gpio_read()` that won't work with `bool gpio_read()` anymore at the top of my head.

### Issues/PRs references

Same that https://github.com/RIOT-OS/RIOT/pull/20935 did for `gpio_write()`, but for `gpio_read()`.

dylad commented on drivers/include/periph/gpio.h:
```diff
@@ -227,10 +227,10 @@ void gpio_irq_disable(gpio_t pin);
- * @return              0 when pin is LOW
- * @return              >0 for HIGH
+ * @retval              false   pin is LOW
+ * @retval              high    pin is HIGH
```
```suggestion
 * @retval              true    pin is HIGH
```

maribu commented:
See https://github.com/RIOT-OS/rust-riot-wrappers/pull/132

dylad reviewed (approved):
Changes look good to me.

dylad commented:
Don't forget to remove the extra commit :)

maribu commented:
> Don't forget to remove the extra commit :)

Done. But since this requires recently merged changes to the rust wrappers to work with both old and new `gpio_read()` signature, I updated the `riot-rust-wrappers` and added this as commit using:

```sh
for dir in $(dirname $(find . -name Cargo.lock)); do 
    pushd $dir
    cargo update riot-wrappers
    popd
done
```

chrysn reviewed (approved):
LGTM at a glance, and I checked the Rust changes to be really just what is needed here.

In general I softly prefer doing the Rust changes outside the breaking PR (so that not just the riot-wrappers' limited CI runs, but it gets a full murdock run), but the change is so small here that doing it in the breaking PR is just as fine.

dylad commented:
@benpicco Should this PR go into master ?

maribu commented:
Thx :)
